packages/common/src/interfaces/resizerOption.interface.ts

```diff
@@ -33,4 +39,18 @@ export interface ResizerOption {
 
   /** padding on the right side of the grid (pixels) */
   rightPadding?: number;
+
+  /**
+   * Use a [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) to detect resizes on the {@link container} element instead of window resize events.
+   *
+   * Remarks:
+   * * Requires {@link container} to be set.
+   *
+   * * ResizeObserver is not supported on older browsers like Internet Explorer. If you need the support you need to install a
+   * polyfill like [resize-observer-polyfill](https://www.npmjs.com/package/resize-observer-polyfill) yourself.
+   *
+   * * If you get 'ResizeObserver loop limit exceeded' errors in automated tests take a look
+   * [here](https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded).
+   */
+  useResizeObserver?: boolean;
```

Comment:
is this resize observer only here because of legacy browser? If that is the only reason, I would prefer to drop the flag and simply make the Resize Observer the default because I'm already no longer supporting legacy browser, not even IE11 because I'm bundling for ES2018 (as shown in [tsconfig.json](https://github.com/ghiscoding/slickgrid-universal/blob/master/tsconfig.json) and so IE11 and lower are excluded. So if this Resize Observer is really better than the window resize event, I would prefer to only keep that one and get rid of the old window event. What do you think?

Reply:
The reason for using a `ResizeObserver` is that it detects resizes on a specific element it observes (which I set to be the `resizerOption.container`). Whereas the window resize only detects entire window resizes. So for example if I just change the container's width (and not the window width) the window resize wouldn't be triggered (and jQuery wouldn't detect a resize) , but `ResizeObserver` does.

I'll change `useResizeObserver` to `resizeDetection?: 'container' | 'window'`,  similar to `calculateAvailableSizeBy`, that should make it more clear.
And I'll remove the comment about old IE and polyfill.

When `resizerOption.container` is set, probably `resizeDetection='container'` would be a good default, but I don't want to change current behavior and break things.

Reply:
yeah as long as we don't have regression I'm fine with that, I was just thinking in a way that might end up with less lines of code.